loadbalancing.md

```diff
@@ -0,0 +1,345 @@
+```
+
+And we need to start HAProxy
+
+```
+sudo systemctl enable haproxy --now
+```
+
+See also [Best Practices for Using HAProxy in TiDB](/best-practices/haproxy-best-practices.md)
+
+### MySQL Router
+
+We first install MySQL Router. Follow the [instructons from the documentaton](https://dev.mysql.com/doc/mysql-router/8.0/en/mysql-router-installation.html) for your platform of choice. For this example I'm using Linux.
+
+```
+sudo rpm -ivh https://dev.mysql.com/get/mysql80-community-release-fc34-1.noarch.rpm
+sudo dnf install mysql-router
+```
+
+Now we need to configure the router by adding the following to `/etc/mysqlrouter/mysqlrouter.conf`:
+
+```
+[routing:tidb]
+bind_address = 127.0.0.1:6446
+routing_strategy = round-robin
+protocol = classic
+destinations = 127.0.0.1:4000, 127.0.0.1:4001
+```
+
+```
+sudo systemctl enable  mysqlrouter --now
+```
+
+Note that you need to use `protocol = classic` as TiDB does not support the [X Protocol](https://github.com/pingcap/tidb/issues/1109).
+
+### MySQL Connector/J
+
+MySQL Connector/J supports various kinds of [Multi-Host Connections](https://dev.mysql.com/doc/connectors/en/connector-j-multi-host-connections.html). In this example we use the `loadbalance` option as that distributes the load over multiple TiDB serves where the `failover` tries to use the first host from the list, resulting in an un-even distribution of load.
+
+The `X Dev API` connection type of Connector/J can not be used with TiDB as [TiDB doesn't support the X Protocol](https://github.com/pingcap/tidb/issues/1109).
+
+An example with MySQL Connector/J 8.0.24:
+
+```
+package com.pingcap.tidb_demo_java;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.sql.ResultSet;
+
+public class App
+{
+    public static void main( String[] args )
+    {
+        try {
+                Class.forName("com.mysql.cj.jdbc.Driver").newInstance();
+        } catch (Exception ex) {
+                System.out.println("Failed to load driver: " + ex.getMessage());
+        }
+
+        try {
+                Connection conn = DriverManager.getConnection("jdbc:mysql:loadbalance://127.0.0.1:4000,127.0.0.1:4001?user=root");
+                Statement stmt = conn.createStatement();
+                ResultSet rs = stmt.executeQuery("SELECT @@hostname, @@port");
+                rs.next();
+                System.out.println(rs.getString("@@hostname") + ":" + rs.getString("@@port"));
+        } catch (SQLException ex) {
+                System.out.println("SQLException: " + ex.getMessage());
+        }
+    }
+}
+```
+
+See [Best Practices for Developing Java Applications with TiDB](/best-practices/java-app-best-practices.md) for more details about how to use TiDB with Java Applications.
+
+### MySQL Connector/Python
+
+In this example I use MySQL Connector/Python 8.0.24 (`mysql-connector-python==8.0.24` according to `pip freeze` output).
```

Comment:
IMHO It's better to avoid the first-person pronoun "I" in this document. Could you use the passive voice here like:

"In this example MySQL Connector/Python 8.0.24 (`mysql-connector-python==8.0.24` according to `pip freeze` output) are used."